### DISCUSSION TITLE
MotionControllerInterface control frequency

### DISCUSSION THREAD

**Discussion by olivierp99** [Yaskawa-Global/YMConnect#9] [General]
Hello!

I was wondering what was the frequency for MotionManagerInterface? Can I stream at 125Hz? can I send all the points at once? what's the buffer capacity?

Great work by the way, the library looks great

**Comment by ted-miller** [comment]
Hello @olivierp99 
YMConnect is not intended for deterministic real-time control at a low level. It is intended for queuing sparse-ish waypoints. I like to compare it to building a teach pendant job on-the-fly. YMConnect is intended for ease of use.

When you submit points to the MotionManager interface, they will be queued up in memory before being transmitted to the robot controller. There is no size limitation to this queue. You can queue them up as fast as your computer will allow. 

Once points are in the library's queue, the library spins up a background thread which manages transmission to the robot controller. As points are processed and the robot's buffer has free space, the library will dequeue your points and send to the robot. These will go into the motion-planner's buffer and a smooth trajectory will be plotted.

If you are looking for more low-level control, I recommend either the MotoROS/MotoROS2 interface or a custom application developed using MotoPlus SDK.
A brief overview is available at https://developer.motoman.com/en/home

**Reply by olivierp99** [reply]
if a use a motoplus sdk app to stream like with motoros can I still use YMConnect or they will interfere?

**Reply by ted-miller** [reply]
Yes, these can be used concurrently. The only conflict would be if you attempt to control motion from both interfaces. One of them would be rejected and fail.
